packages/contracts/src/CardsCollection.sol

```diff
@@ -81,6 +85,10 @@ contract CardsCollection is ERC721, Ownable {
     }
 
     function getCard(uint256 card) external view returns (Card memory) {
-        return Card(card, lore[card], stats_[card]);
+        return Card(card, lore[card], stats_[card], cardType[card]);
+    }
+ 
+    function getCardType(uint256 card) external view returns(uint32){
+        return cardType[card];
     }
```

Comment:
This is redundant if the `cardType` mapping is public: Solidity will generate a function called `cardType` that other contract can call that acts exactly like this.

It's necessary for `getLore` etc because Solidity is weird and the autogenerated function returns a tuple of values instead of the struct type (and you can't assign the tuple directly to the struct type, that would be way too easy).

Reply:
right!